app/kubernetes/views/applications/applicationsController.js

```diff
@@ -102,8 +109,16 @@ class KubernetesApplicationsController {
 
   async getApplicationsAsync() {
     try {
-      const applications = await this.KubernetesApplicationService.get();
-      this.applications = applications;
+      const [applications, configurations] = await Promise.all([this.KubernetesApplicationService.get(), this.KubernetesConfigurationService.get()]);
+      const configuredApplications = KubernetesConfigurationHelper.getApplicationConfigurations(applications, configurations);
+      const helmApplications = KubernetesApplicationHelper.getHelmApplications(configuredApplications);
+
+      // filter out multi-chart helm managed applications
+      const singularApplications = configuredApplications.filter(
+        (app) => !app.Pods.flatMap((pod) => pod.Labels).some((label) => label && helmApplications.map((hma) => hma.Name).includes(label['app.kubernetes.io/instance']))
```

Comment:
move `elmApplications.map((hma) => hma.Name)` to line 116, so it won't run it on every label.
also, why not:
```
const helmNames = helmApplications.map((hma) => hma.Name);

const singularApplications = configuredApplications.filter((app) =>
  !app.Pods.flatMap((pod) =>
    pod.Labels.filter((l) => l).map((l) => l["app.kubernetes.io/instance"])
  ).some((label) => helmNames.includes(label))
);
```

it's more readable for me. I'm not sure why it's considered `singular` maybe just `nonHelmApps`?

Reply:
bad at naming 😮‍💨 

great suggestions 👍